pump.go

```diff
@@ -108,10 +110,10 @@ func (p *asyncPump) run() {
 		start := nanotime()
 		err := p.processor.Process(msg)
 		if err != nil {
-			p.Unlock()
-			p.errFn(err)
-
-			return
+			if err = p.errFn(err); err != nil {
```

Comment:
One thing I see here is Unlock function call is now optional but I guess its intended. The other difference is the p.errFn is now called before p.Unlock. Good, as long as you know its ok.

Reply:
Yeah, previously `p.Unlock` has been called before `errFn`, but we were always stopping the task. Currently task is supposed to continue running until you return an error from `errFn`.